code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeUtils.java

```diff
@@ -175,4 +179,164 @@ static String getRequestEventId(final Event event) {
+                            }
+                        });
+            }
+        }
+        final Map<String, Object> experienceDecisioning = new HashMap<>();
+        experienceDecisioning.put(
+                OptimizeConstants.JsonKeys.DECISIONING_PROPOSITIONS, decisioningPropositions);
+
+        // Create experience map
+        final Map<String, Object> experience = new HashMap<>();
+        experience.put(OptimizeConstants.JsonKeys.EXPERIENCE_DECISIONING, experienceDecisioning);
+
+        // Create xdm map
+        final Map<String, Object> xdm = new HashMap<>();
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE, experience);
+        xdm.put(OptimizeConstants.JsonKeys.EXPERIENCE_EVENT_TYPE, experienceEventType);
+
+        return xdm;
+    }
+
+    /**
+     * Dispatches an event to track propositions with type {@value
+     * OptimizeConstants.EventType#OPTIMIZE} and source {@value
+     * OptimizeConstants.EventSource#REQUEST_CONTENT}.
+     *
+     * <p>No event is dispatched if the provided {@code xdm} is null or empty.
+     *
+     * @param xdm {@code Map<String, Object>} containing the XDM data for the proposition
+     *     interactions.
+     */
+    static void trackWithData(final Map<String, Object> xdm) {
+        if (OptimizeUtils.isNullOrEmpty(xdm)) {
+            Log.debug(
+                    OptimizeConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Failed to dispatch track propositions request event, input xdm is null or"
+                            + " empty.");
+            return;
+        }
+
+        final Map<String, Object> eventData = new HashMap<>();
+        eventData.put(
+                OptimizeConstants.EventDataKeys.REQUEST_TYPE,
+                OptimizeConstants.EventDataValues.REQUEST_TYPE_TRACK);
+        eventData.put(OptimizeConstants.EventDataKeys.PROPOSITION_INTERACTIONS, xdm);
+
+        final Event edgeEvent =
+                new Event.Builder(
+                                OptimizeConstants.EventNames.TRACK_PROPOSITIONS_REQUEST,
+                                OptimizeConstants.EventType.OPTIMIZE,
+                                OptimizeConstants.EventSource.REQUEST_CONTENT)
+                        .setEventData(eventData)
+                        .build();
+
+        MobileCore.dispatchEvent(edgeEvent);
+    }
+
+    /**
+     * Generates a map containing XDM formatted data for `Experience Event - OptimizeProposition
+     * Interactions` field group from this `OptimizeProposition` offer and given
+     * `experienceEventType`.
+     *
+     * <p>
+     *
+     * <p>The method returns null if the proposition reference within the offer is released and no
+     * longer valid.
+     *
+     * @param experienceEventType [String] containing the event type for the Experience Event
+     * @return `Map<String, Object>` containing the XDM data for the proposition interaction.
+     */
+    protected static Map<String, Object> generateInteractionXdm(
```

Comment:
What is the difference between this one and the other `generateInteractionXdm`?

Reply:
removed @spoorthipujariadobe